modules/claims_api/app/models/claims_api/form_526.rb

```diff
@@ -68,10 +69,61 @@ def to_internal
       }.to_json
     end
 
+    def current_mailing_address
+      veteran[:currentMailingAddress]
+    end
+
     private
 
     def cleaned_errors
       errors.messages.map { |key, value| "#{key} " + value.join(' and ') }.join(', ')
     end
+
+    def validate_nested_inputs
+      validate_veteran
+      validate_service_information
+      validate_disabilities
+      validate_direct_deposit if directDeposit.present?
+    end
+
+    def validate_veteran
+      key = current_mailing_address
+      errors.add(:veteran, 'must include currentMailingAddress') unless veteran.key?(:currentMailingAddress)
+      errors.add(:currentMailingAddress, 'currentMailingAddress is not an object') unless key.is_a?(Hash)
+      validate_current_mailing_address unless errors[:currentMailingAddress].any?
+    end
+
+    def validate_current_mailing_address
+      attributes = %i[addressLine1 city state zipFirstFive country type]
+      validate_attribute_set(attributes, current_mailing_address, 'currentMailingAddress')
+    end
+
+    def validate_direct_deposit
+      attributes = %i[accountType accountNumber routingNumber]
+      validate_attribute_set(attributes, directDeposit, 'directDeposit')
+    end
+
+    def validate_attribute_set(keys, parent, label)
+      keys.each do |key|
+        errors.add(label, "must include #{key}") unless parent.key?(key)
+      end
+    end
```

Comment:
Renaming this to `validate_keys_set` and using kwargs would more clearly communicate to me what's happening here. The singular wording made me wonder why a list of fields was being passed in when I saw the method call above. In models I expect `attribute` to refer to an accessor method that rails added to reference a field in the underlying data store. With these changes the call above would change:

```
-       validate_attribute_set(attributes, current_mailing_address, 'currentMailingAddress')
+       validate_keys_set(keys: attributes, in: current_mailing_address, errorLabel: 'currentMailingAddress')
```